File: DataSets/DataSet-New/LeetCodeTasks - Develop/LeetCodeTasks/3091_apply-operations-to-make-sum-of-array-greater-than-or-equal-to-k/solutions/java/solution.java
class Solution {
    public int minOperations(int k) {
        if (k == 1) return 0;

        int sum = 1;
        int ops = 0;
        int maxElement = 1;

        while (sum < k) {
            if (sum + maxElement <= k) {
                sum += maxElement;
                ops++; // This operation was a duplication
            } else {
                int needed = k - sum;
                maxElement += needed;
                sum += needed;
                ops += needed; // These operations are increment operations
                break;
            }
        }

        return ops;
    }
}

File: DataSets/DataSet-New/LeetCodeTasks - Develop/LeetCodeTasks/3091_apply-operations-to-make-sum-of-array-greater-than-or-equal-to-k/solutions/c/solution.c
#include <limits.h>

int minOperations(int k) {
    if (k == 1) return 0;
    
    int sum = 1;
    int count = 0;
    
    // The optimal way to increase the sum quickly is to duplicate the largest values.
    while (sum < k) {
        int needed = k - sum;
        if (sum >= needed) {
            sum += sum;  // Doubling the sum.
            count++;     // One operation to double the highest possible value (or values).
        } else {
            int maxIncrement = INT_MAX - sum;
            if (maxIncrement > needed) {
                sum += needed;
                count += needed;  // Individual increments to the highest possible value.
            } else {
                sum += maxIncrement;
                count += maxIncrement;
            }
        }
    }
    
    return count;
}
